### PR TITLE
Split Travis jobs into two machines (parallel tasks)

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal: Split Travis jobs into two machines (parallel tasks). Closes cksource/ckeditor5-internal#706.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by Reinmar** [comment]
Naming proposal:

*   Coverage -> Tests
*   Validate -> Validation

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,27 @@
...
+# "travis_wait" does not work in child processes. Hence, it must be called from the configuration file.
+- if [[ $TRAVIS_JOB_TYPE == "Validate" && $TRAVIS_TEST_RESULT -eq 0 ]]; then travis_wait 30 yarn run docs:build-and-publish-nightly; fi
 after_script:
 - export END_TIME=$( date +%s )
 - ckeditor5-dev-tests-notify-travis-status
 env:
-  global:
-  - secure: RO140EQDHmEOPJPikk8eCY5IdHpnEKGm41p5U1ewAbeZv1DpCG+rSumR2JdYl75kFAaZvCSm1NuVMM+kmYd+/z+LQbKj7QH5G/UHNho3H89blIU6WlJhT0YR5vclm9rvnEvOtxnfODca1Qrw+CaCoJks2o4VYbJB7mOBVNsh7Bc=
+  jobs:
+  - TRAVIS_JOB_TYPE=Coverage
```
Comment (pomek) [inline .travis.yml:26]:
```suggestion
  - TRAVIS_JOB_TYPE=Tests
```

File: scripts/ci/travis-check.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+const { TRAVIS_JOB_TYPE } = process.env;
+
+const RED = '\x1B[0;31m';
+const GREEN = '\x1B[32m';
+const CYAN = '\x1B[36;1m';
+const MAGENTA = '\x1B[35;1m';
+const NO_COLOR = '\x1B[0m';
+
+// Tests + Code coverage.
+if ( TRAVIS_JOB_TYPE === 'Coverage' ) {
```
Comment (pomek) [inline scripts/ci/travis-check.js:26]:
```suggestion
if ( TRAVIS_JOB_TYPE === 'Tests' ) {
```

File: scripts/ci/travis-check.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	console.log( `\n${ CYAN }Running the "Coverage" build.${ NO_COLOR }\n` );
+
+	exec( 'node', './scripts/ci/check-packages-code-coverage.js' );
+}
+
+// Verifying the code style.
+if ( TRAVIS_JOB_TYPE === 'Validate' ) {
```
Comment (pomek) [inline scripts/ci/travis-check.js:33]:
```suggestion
if ( TRAVIS_JOB_TYPE === 'Validation' ) {
```

File: scripts/ci/travis-check.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	console.log( `\n${ CYAN }Running the "Validate" build.${ NO_COLOR }\n` );
```
Comment (pomek) [inline scripts/ci/travis-check.js:34]:
```suggestion
	console.log( `\n${ CYAN }Running the "Validation" build.${ NO_COLOR }\n` );
```

File: scripts/ci/travis-check.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+const RED = '\x1B[0;31m';
+const GREEN = '\x1B[32m';
+const CYAN = '\x1B[36;1m';
+const MAGENTA = '\x1B[35;1m';
+const NO_COLOR = '\x1B[0m';
+
+// Tests + Code coverage.
+if ( TRAVIS_JOB_TYPE === 'Coverage' ) {
+	console.log( `\n${ CYAN }Running the "Coverage" build.${ NO_COLOR }\n` );
```
Comment (pomek) [inline scripts/ci/travis-check.js:27]:
```suggestion
	console.log( `\n${ CYAN }Running the "Tests" build.${ NO_COLOR }\n` );
```

File: .travis.yml [hunk 6]
```diff
@@ -1,37 +1,27 @@
...
+  - TRAVIS_JOB_TYPE=Validate
```
Comment (pomek) [inline .travis.yml:27]:
```suggestion
  - TRAVIS_JOB_TYPE=Validation
```

File: .travis.yml [hunk 7]
```diff
@@ -1,37 +1,27 @@
...
-- yarn run lint
-- yarn run stylelint
-- ./scripts/check-manual-tests.sh -r ckeditor5 -f ckeditor5
-- yarn run docs --strict
-- 'if [ $TRAVIS_TEST_RESULT -eq 0 ]; then
-    travis_wait 30 yarn run docs:build-and-publish-nightly;
-  fi'
+- ./scripts/ci/travis-check.js
+# "travis_wait" does not work in child processes. Hence, it must be called from the configuration file.
+- if [[ $TRAVIS_JOB_TYPE == "Validate" && $TRAVIS_TEST_RESULT -eq 0 ]]; then travis_wait 30 yarn run docs:build-and-publish-nightly; fi
```
Comment (pomek) [inline .travis.yml:20]:
```suggestion
- if [[ $TRAVIS_JOB_TYPE == "Validation" && $TRAVIS_TEST_RESULT -eq 0 ]]; then travis_wait 30 yarn run docs:build-and-publish-nightly; fi
```

**Review by pomek — COMMENTED** [review]
Wording.

**Review by psmyrek — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**